Sources/Internal.swift

```diff
@@ -260,6 +260,7 @@ struct ResumableData {
         // Check if "Accept-Ranges" is present and the response is valid.
         guard !data.isEmpty,
             let response = response as? HTTPURLResponse,
+            data.count < response.expectedContentLength,
```

Comment:
URLSession always seem to return cached data in a single chunk, so this should always prevent cached data from being saved as resumable data. Ideally, I would like to explicitly check if data is coming from cache, but this doesn't seem to be possible.

Reply:
This should also cover the following (rate) scenario:

- Data fully downloaded form the server
- `func imageDataLoadingTask(_ task: OriginalImageDataTask, context: OriginalImageDataTaskContext, didReceiveData chunk: Data, response: URLResponse, log: Log)` is called with the last chunk
- The request is cancelled and resumable data is saved
- `func imageDataLoadingTask(_ task: OriginalImageDataTask, context: OriginalImageDataTaskContext, didFinishLoadingDataWithError error: Swift.Error?)` is called